axelor-contract/src/main/java/com/axelor/apps/contract/web/ContractLineController.java

```diff
@@ -59,4 +75,157 @@ public void createAnalyticDistributionWithTemplate(
+    try {
+      if (request.getContext().getParent() != null
+          && (Contract.class)
+              .equals(request.getContext().getParent().getParent().getContextClass())) {
+
+        ContractLine contractLine = request.getContext().asType(ContractLine.class);
+        Contract contract = request.getContext().getParent().getParent().asType(Contract.class);
+        if (contract != null
+            && Beans.get(MoveLineComputeAnalyticService.class)
+                .checkManageAnalytic(contract.getCompany())) {
+          contractLine =
+              Beans.get(ContractLineService.class)
+                  .analyzeContractLine(contractLine, contract, contract.getCompany());
+          response.setValue("analyticMoveLineList", contractLine.getAnalyticMoveLineList());
+        }
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void manageAxis(ActionRequest request, ActionResponse response) {
+    try {
+      Contract contract = null;
+
+      if (request.getContext().getParent() != null
+          && (Contract.class)
+              .equals(request.getContext().getParent().getParent().getContextClass())) {
+        contract = request.getContext().getParent().getParent().asType(Contract.class);
+      }
+
+      if (contract != null && contract.getCompany() != null) {
+        AccountConfig accountConfig =
+            Beans.get(AccountConfigService.class).getAccountConfig(contract.getCompany());
+        if (Beans.get(MoveLineComputeAnalyticService.class)
+            .checkManageAnalytic(contract.getCompany())) {
+          AnalyticAxis analyticAxis = null;
+          for (int i = startAxisPosition; i <= endAxisPosition; i++) {
+            response.setAttr(
+                "axis".concat(Integer.toString(i)).concat("AnalyticAccount"),
+                "hidden",
+                !(i <= accountConfig.getNbrOfAnalyticAxisSelect()));
+            for (AnalyticAxisByCompany analyticAxisByCompany :
+                accountConfig.getAnalyticAxisByCompanyList()) {
+              if (analyticAxisByCompany.getSequence() + 1 == i) {
+                analyticAxis = analyticAxisByCompany.getAnalyticAxis();
+              }
+            }
+            if (analyticAxis != null) {
+              response.setAttr(
+                  "axis".concat(Integer.toString(i)).concat("AnalyticAccount"),
+                  "title",
+                  analyticAxis.getName());
+              analyticAxis = null;
+            }
+          }
+        } else {
+          response.setAttr("analyticDistributionTemplate", "hidden", true);
+          response.setAttr("analyticMoveLineList", "hidden", true);
+          for (int i = startAxisPosition; i <= endAxisPosition; i++) {
+            response.setAttr(
+                "axis".concat(Integer.toString(i)).concat("AnalyticAccount"), "hidden", true);
+          }
+        }
+      }
```

Comment:
same issue as the other methods, there are too much service related code inside these methods.